composer/admin.py

```diff
@@ -28,9 +33,127 @@ def __init__(self, *args, **kwargs):
         except (AttributeError, KeyError):
             styles = []
         styles.append(("tile", "Tile"))
+        styles = self.get_existing_styles(styles)
         styles.sort()
         self.fields["style"].widget = forms.widgets.Select(choices=styles)
 
+    def get_existing_styles(self, current_styles):
+        # Return the current styles if the composer setting or the specific key
+        # is missing.
+        try:
+            setting = settings.COMPOSER
+        except (AttributeError):
+            return current_styles
+
+        if not setting.get("load-existing-styles"):
+            return current_styles
+
+        # Get a list of all the availible templates in the project.
+        styles_settings = setting["load-existing-styles"]
+        greedy = styles_settings.get("greedy", False)
+        excludes = styles_settings.get("excludes")
+        includes = styles_settings.get("includes")
+
+        # If for some reason none of the actual values are supplied return an
+        # empty list.
+        if not greedy and excludes is None and includes is None:
```

Comment:
Parenthesis makes it easier to read. `if not greedy and (excludes is None) and (includes is None)`